firebase-firestore/src/androidTest/java/com/google/firebase/firestore/AggregationTest.java

```diff
@@ -0,0 +1,1030 @@
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(collection.aggregate(average("pages")).get(AggregateSource.SERVER));
+
+    assertEquals((Double) 75.0, snapshot.getDouble(average("pages")));
+
+    AggregateQuerySnapshot snapshotWeight =
+        waitFor(collection.aggregate(average("weight")).get(AggregateSource.SERVER));
+
+    assertEquals((Long) 49L, snapshotWeight.getLong(average("weight")));
+    assertEquals((Double) 49.8, snapshotWeight.getDouble(average("weight")));
+  }
+
+  @Test
+  public void testCanGetDuplicateAggregations() {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(
+            collection
+                .aggregate(
+                    AggregateField.count(), AggregateField.count(), sum("pages"), sum("pages"))
+                .get(AggregateSource.SERVER));
+
+    assertEquals(2L, snapshot.get(AggregateField.count()));
+    assertEquals(150L, snapshot.get(sum("pages")));
+  }
+
+  @Test
+  public void testCanGetMultipleAggregationsInTheSameQuery() {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(
+            collection
+                .aggregate(AggregateField.count(), sum("pages"), average("pages"))
+                .get(AggregateSource.SERVER));
+
+    assertEquals(2L, snapshot.get(AggregateField.count()));
+    assertEquals(150L, snapshot.get(sum("pages")));
+    assertEquals((Double) 75.0, snapshot.getDouble(average("pages")));
+  }
+
+  @Test
+  public void testTerminateDoesNotCrashWithFlyingAggregateQuery() {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+
+    collection
+        .aggregate(AggregateField.count(), sum("pages"), average("pages"))
+        .get(AggregateSource.SERVER);
+
+    waitFor(collection.firestore.terminate());
+  }
+
+  @Test
+  public void testCanGetCorrectTypeForSum() {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(
+            collection
+                .aggregate(sum("pages"), sum("height"), sum("weight"))
+                .get(AggregateSource.SERVER));
+
+    Object sumPages = snapshot.get(sum("pages"));
+    Object sumHeight = snapshot.get(sum("height"));
+    Object sumWeight = snapshot.get(sum("weight"));
+    assertTrue(sumPages instanceof Long);
+    assertTrue(sumHeight instanceof Long);
+    assertTrue(sumWeight instanceof Double);
+  }
+
+  @Test
+  public void testCanGetCorrectTypeForAvg() {
+    CollectionReference collection = testCollectionWithDocs(testDocs1);
+
+    AggregateQuerySnapshot snapshot =
+        waitFor(collection.aggregate(average("pages")).get(AggregateSource.SERVER));
+
+    Object avg = snapshot.get(average("pages"));
+    assertTrue(avg instanceof Double);
+  }
+
+  @Test
+  public void testCanPerformMaxAggregations() {
```

Comment:
These MaxAggregations tests will not validate anything because of de-duplication logic. We will need to add them back in the future.

Reply:
I don't understand why the de-duplication logic would have impact on this test? the aggregate field inside is different.